sdk/keyvault/keyvault-keys/src/lro/delete/poller.ts

```diff
@@ -0,0 +1,55 @@
+import { delay, RequestOptionsBase } from "@azure/core-http";
+import { Poller } from "@azure/core-lro";
+import { DeleteKeyPollOperationState, makeDeleteKeyPollOperation } from "./operation";
+import { DeletedKey, KeyClientInterface } from "../../keysModels";
+
+export interface DeleteKeyPollerOptions {
+  client: KeyClientInterface;
+  name: string;
+  requestOptions?: RequestOptionsBase;
+  intervalInMs?: number;
+  resumeFrom?: string;
+}
+
+/**
+ * Class that deletes a poller that waits until a key finishes being deleted
+ */
+export class DeleteKeyPoller extends Poller<DeleteKeyPollOperationState, DeletedKey> {
+  /**
+   * Defines how much time the poller is going to wait before making a new request to the service.
+   * @memberof DeleteKeyPoller
+   */
+  public intervalInMs: number;
+
+  constructor(options: DeleteKeyPollerOptions) {
+    const { client, name, requestOptions, intervalInMs = 1000, resumeFrom } = options;
+
+    let state: DeleteKeyPollOperationState | undefined;
+
+    if (resumeFrom) {
+      state = {
+        ...JSON.parse(resumeFrom).state,
+        ...state
```

Comment:
what does this line do? state should be undefined at this point if I'm not mistaken

Reply:
I'll get rid of it!